tests/unit/experiments/providers/simple_experiment_tests.py

```diff
@@ -417,3 +444,148 @@ def test_targeting_in_config(self):
             is_logged_in=True,
             random_numeric=5,
         ))
+
+    def test_construct_override(self):
+        cfg = get_simple_override_config()
+        overrides = _generate_overrides(cfg)
+
+        self.assertEqual(len(overrides), 4)
+        
+        override_names_and_types = [
+            ('override_variant_1', EqualNode),
+            ('override_variant_2', EqualNode),
+            ('override_variant_3', EqualNode),
+            ('override_variant_1', EqualNode),
+        ]
+
+        for i in range(4):
+            self.assertEqual(len(overrides[i]),1)
+            self.assertTrue(isinstance(
+                overrides[i][override_names_and_types[i][0]],
+                override_names_and_types[i][1]
+            ))
```

Comment:
I think it would be simpler to do this as:

```
for i, override in enumerate(overrides):
    self.assertEqual(len(override), 1)
    variant_name, override_type = override_names_and_types[i]
    self.assertTrue(isinstance(override[variant_name], override_type)
```

or something like that

Reply:
TIL about `enumerate`. Will update.